### PR TITLE
Reserve protection et al

### PR DESCRIPTION
**Pull request by austinhuang0131**
Please Prettier I cannot, but works it indeed

### CODE REVIEW

File: src/transactions/conversion-check.guard.ts [hunk 1]
```diff
@@ -1,29 +1,46 @@
...
 
 		if (body) {
+			const toCurrency = await currencies.findOne(body.toId);
 			if (body.toId === signedInBot.currency.id) {
 				throw new BadRequestException(
 					`You can not convert ${signedInBot.currency.id} to ${body.toId} because they are the same`
 				);
+			} else if (!toCurrency) {
+				throw new BadRequestException(`Currency ${body.toId} does not exist`);
+			} else if ((body.amount * signedInBot.currency.value) / toCurrency.value >= toCurrency.reserve) {
```
Comment (pizzafox) [inline src/transactions/conversion-check.guard.ts]: This is assuming that the provided body actually has an `amount` property.

File: src/transactions/conversion-check.guard.ts [hunk 2]
```diff
@@ -1,29 +1,46 @@
...
+	BadRequestException,
+	ForbiddenException
+} from '@nestjs/common';
+import {getRepository} from 'typeorm';
 import {SignedInBot} from 'types/bot';
+import {Currency} from 'src/currencies/currency.entity';
 import {Transaction} from './transaction.entity';
 
 @Injectable()
 export class ConversionCheckGuard implements CanActivate {
```
Comment (pizzafox) [inline src/transactions/conversion-check.guard.ts]: Also, all this added logic should be in a validator file, not a guard. Relevant docs on [validation](https://docs.nestjs.com/techniques/validation) with `class-validator` and broader use [validation pipes](https://docs.nestjs.com/pipes) (more suited for this use case).

File: src/transactions/conversion-check.guard.ts (hunk 1 above)
Comment (austinhuang0131) [reply src/transactions/conversion-check.guard.ts]: https://github.com/Discoin/api-v3/commit/c4dc2edc7d102cdbf229deb18dbcfa6fdc645690

**Comment by austinhuang0131** [comment]
for personal info: reserve should not go over 1 bil

**Comment by Rexogamer** [comment]
anything new on this

**Comment by austinhuang0131** [comment]
will do this on a new branch

**Pull request closed without merging**